front/app/api/background_jobs/useTrackBackgroundJobs.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+import { IBackgroundJobData } from './types';
+import useBackgroundJobs from './useBackgroundJobs';
+
+const useTrackJobs = ({
+  jobs,
+  onChange,
+}: {
+  jobs: IBackgroundJobData[];
+  onChange: () => void;
+}) => {
+  const trackedJobsIds = jobs.map((job) => job.attributes.job_id);
+
+  const { data: polledJobs } = useBackgroundJobs(trackedJobsIds);
+
+  useEffect(() => {
+    onChange();
+  }, [polledJobs?.data, onChange]);
+
+  const active = !!polledJobs?.data.some((job) => job.attributes.active);
+  const failed =
+    polledJobs?.data &&
+    polledJobs.data.length > 0 &&
+    polledJobs.data.every((job) => job.attributes.failed);
```

Comment:
We could maybe add a comment to explain that successful jobs are removed, hence the `every`.

Reply:
Makes sense, done https://github.com/CitizenLabDotCo/citizenlab/pull/7704/commits/b48c8d243eb43629c5838fb726d59042b2889d94